code/iaas/logic/src/main/java/io/cattle/platform/process/ipaddress/IpAddressActivate.java

```diff
@@ -54,6 +62,29 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
                 : ipAddress.getName(), IP_ADDRESS.NETWORK_ID, networkId);
     }
 
+    protected PooledResourceOptions getPoolOptions(IpAddress ipAddress) {
+        PooledResourceOptions options = new PooledResourceOptions();
+
+        if (IpAddressConstants.ROLE_PRIMARY.equals(ipAddress.getRole())) {
+            for (Nic nic : getObjectManager().mappedChildren(ipAddress, Nic.class)) {
+                if (nic.getDeviceNumber() != null && nic.getDeviceNumber() == 0) {
+                    Instance instance = getObjectManager().loadResource(Instance.class, nic.getInstanceId());
+                    if (instance == null) {
+                        continue;
+                    }
+
+                    String ip = DataAccessor.fieldString(instance, InstanceConstants.FIELD_REQUESTED_IP_ADDRESS);
+                    if (ip != null) {
+                        options.setRequestedItem(ip);
+                        break;
+                    }
+                }
```

Comment:
Darren, can ipAddress belong to more than one instance? If not, should we break outside of "if" block, as there is no reason to iterate further even if the ip param is null?